lib/pages/portal/service/website_provider.dart

```diff
@@ -103,6 +104,47 @@ class WebsiteProvider with ChangeNotifier {
     }
   }
 
+  ///This method initializes the number of visits of websites with the value
+  /// accumulated until then. These data are stored in 2 lists: the list of website ids
+  /// and the list with the number of accesses, according to the index in the list
+  void initializeNumberOfVisits(List<Website> websites) {
+    List<String> websiteIds =
+        PrefService.sharedPreferences.getStringList("websiteIds");
+    List<String> visitsByWebsiteId =
+        PrefService.sharedPreferences.getStringList("visitsByWebsiteId");
+
+    if (websiteIds != null && visitsByWebsiteId != null) {
+      var idsMap = Map<String, int>.from(websiteIds.asMap().map((index, key) =>
+          MapEntry(key, int.tryParse(visitsByWebsiteId[index] ?? 0))));
+      websites.forEach((website) {
+        if (idsMap[website.id] != null) {
+          website.numberOfVisits = idsMap[website.id];
+        }
+      });
+    }
+  }
+
+  ///this method increment the number of visits of websites when it is accessed
+  /// by the user. In case the site has not been saved until now, it creates
+  /// another element in the 2 lists(id and visits) for the respective site
```

Comment:
```suggestion
  /// Increments the number of visits of [website], both in-memory and on the local storage.
  ///
  /// Because [PrefService] doesn't support storing maps, the
  /// data is stored in 2 lists: the list of website IDs ([websiteIds]) and the list
  /// with the number of visits ([websiteVisits]), where `websiteVisits[i]` is the 
  /// number of times the user accessed website with ID `websiteIds[i]`.
```